### PR TITLE
Cleanup types

### PR DESCRIPTION
**Pull request by pflopez**
This pr removes a ton of duplicate types and transforms when dealing with plans, fixes  some wrong types, removes a lot of optional types (which are not optional) and updates the codebase to use the "new" properties (basically, the same properties that the BE provides).

### CODE REVIEW

File: src/interface/src/app/services/plan.service.ts [hunk 1]
```diff
@@ -52,125 +46,33 @@ export class PlanService {
...
-            ...plan,
-            id: result['id'].toString(),
-            ownerId: result.ownerId,
-            savedScenarios: 0,
-            area_m2: result.area_m2,
-            area_acres: result.area_acres,
-            creator: result.creator,
-          };
-        })
-      );
```
Comment (lastminutediorama) [inline src/interface/src/app/services/plan.service.ts:167]: oh man, that's some heroic cleanup

File: src/interface/src/app/types/plan.types.ts [hunk 2]
```diff
@@ -1,84 +1,26 @@
...
-  created_at?: string;
-  latest_updated?: string;
-  area_acres?: number;
-  area_m2?: number;
-  creator?: string;
-  role?: string;
-  permissions?: string[];
+  role: string;
+  scenario_count: number;
+  user: number;
```
Comment (lastminutediorama) [inline src/interface/src/app/types/plan.types.ts:17]: thanks for alphabetizing!

**Review by lastminutediorama — APPROVED** [review]
:tada::tada::tada:

File: src/interface/src/app/services/mocks.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  geometry: {
+    type: 'FeatureCollection',
+    features: [],
+  },
+  area_acres: 123,
+  area_m2: 345,
+  creator: 'John Plans',
+  created_at: '2024-03-05T18:08:40.503963Z',
+  latest_updated: '2024-03-05T18:08:40.503963Z',
+  permissions: [],
```
[the remark this answers is not in the document]
Comment (pflopez) [reply src/interface/src/app/services/mocks.ts:17]: I think its better to keep it empty and override in tests if needed

**Pull request merged**